app/controllers/application_controller.rb

```diff
@@ -172,9 +172,27 @@ def forward_to_app_config_domain
     redirect_to URL.url(request.fullpath)
   end
 
+  def current_user
+    return super unless edge_cached_action?
+
+    return nil unless session_current_user_id.present?
+      message = "😱😱😱😱😱 You cannot use current_user in this action\
+  because this code path will be cached at the edge in production.\
+  Check out this section in the docs: https://docs.forem.com/technical-overview/architecture/#we-cache-many-content-pages-on-the-edge"
+      p message
+      Object.const_set("YouCannotUseCurrentUserHereCheckTheTerminalOutput", Class.new)
+  end
+  helper_method :current_user
+
   protected
 
   def configure_permitted_parameters
     devise_parameter_sanitizer.permit(:sign_up, keys: %i[username name profile_image profile_image_url])
   end
+
+  def edge_cached_action?
+    # X-Accel-Expires will only be present if we are setting cache headers
+    # And let's only run this code in development and test
+    !Rails.env.production? && response.headers["X-Accel-Expires"].present?
```

Comment:
**question**: Is there a way we can make this a little more future proof? I suspect how we define what's "cached" will change in the future. For example, this header is only for Nginx though it is currently always set on cached responses. If we change how we're using Nginx caching, this might break.

**suggestion**: I know this a protected method, so maybe some sort of spec to test the behavior higher up would help account for the future updates?